torch/_inductor/codegen/codegen_device_driver.py

```diff
@@ -72,10 +72,14 @@ def cuda_kernel_driver() -> str:
                 ));
             }
     """
+    prop = torch.cuda.get_device_properties(0)
+    gfx_arch = prop.gcnArchName.split(":")[0]
+
     if torch.version.hip is not None:
         # Replace the warp size from 32 (cuLaunchKernel) to 64 (hipModuleLaunchKernel)
         # The warp size on NV GPU is 32, while the wavefront size on AMD GPU is 64
-        source_codes = source_codes.replace("32*numWarps", "64*numWarps")
+        if gfx_arch not in ["gfx1100", "gfx1101", "gfx1102"]:
```

Comment:
```suggestion
        if gfx_arch.startswith("gfx10") or gfx_arch.startswith("gfx11"):
```
Can we make this work for any of our warpSize==32 targets?  Would be nice if in the future device props could return warpSize and not just the arch name, but this is the best we can do today.

Reply:
Don't we have `warpSize` already? `https://rocm.docs.amd.com/projects/HIP/en/latest/doxygen/html/structhip_device_prop__t.html#af3357d33c004608bf05bc21a352be81b`

Reply:
I think we should add `warpSize` to `torch.cuda.get_device_properties()` in this PR itself, similar to how `gcnArchName` was added in https://github.com/pytorch/pytorch/commit/333d5821ee1fa01e1184299cad38a43e6e6ccd08, and then we should upstream that change as well. What do you think, @jeffdaily?

Reply:
I am able to get warp_size as part of the device properties. Let me resubmit PR

```
>>> import torch
>>> prop = torch.cuda.get_device_properties(0)
>>> prop.
prop.gcnArchName                      prop.is_multi_gpu_board               prop.max_threads_per_multi_processor  prop.multi_processor_count            prop.total_memory
prop.is_integrated                    prop.major                            prop.minor                            prop.name                             prop.wrap_size
>>> prop.wrap_size
32
```

Reply:
`wrap_size` ?  misspelled?  I'm not seeing that attribute for latest tip of main branch.

Reply:
> `wrap_size` ? misspelled? I'm not seeing that attribute for latest tip of main branch.

Fixed it :-)